File: Backend/src/main/java/com/espumablanca/backend/application/usecase/RoleService.java
package com.espumablanca.backend.application.usecase;

import com.espumablanca.backend.domain.model.dto.request.role.RoleRequest;
import com.espumablanca.backend.domain.model.dto.request.role.RoleUpdatePermissionsRequest;
import com.espumablanca.backend.domain.model.dto.response.role.RoleDto;

import java.util.List;

public interface RoleService {
    RoleDto create(RoleRequest request);

    RoleDto getById(Long id);

    List<RoleDto> getAll();

    RoleDto updatePermissions(Long id, RoleUpdatePermissionsRequest request);
}

File: Frontend/src/app/core/services/api/role.service.ts
import { HttpClient } from '@angular/common/http';
import { inject, Injectable } from '@angular/core';
import { environment } from '../../../../environments/environment.development';
import { RoleDto } from '../../models/role/response/role-dto.model';
import { RoleUpdatePermissionsRequest } from '../../models/role/request/role-update-permissions-request.model';

@Injectable({
  providedIn: 'root',
})
export class RoleService {
  private readonly endpoint = environment.apiUrl + '/protected/roles';
  private readonly http = inject(HttpClient);

  getAll() {
    return this.http.get<RoleDto[]>(this.endpoint);
  }

  updatePermissions(id: number, request: RoleUpdatePermissionsRequest) {
    return this.http.put<RoleDto[]>(this.endpoint + '/' + id, request);
  }
}
